### PR TITLE
MAYA-121681 allow creating prim in stronger layer

### PR DESCRIPTION
**Pull request by pierrebai-adsk**

### CODE REVIEW

**Comment by pierrebai-adsk** [comment]
Only PF failure is 2022 Windows Python 3, testUsdExportPackage fails to delete a file, which has been seen before (flaky) and is unrelated. All others passed.

File: test/lib/ufe/testGroupCmd.py [hunk 1]
```diff
@@ -581,6 +581,62 @@ def testGroupPreserveLoadRules(self):
...
 
+    def testGroupLiftedRestrictions(self):
+        '''
+        Test that we can create a group in a stronger layer even though
+        the prim has been authored in a weaker layer.
+
+            1. Create a stage with a new layer.
+            2. Add a sub-layer.
+            3. Set the edit target to the new sub-layer.
+            4. Create on xform under the proxyShape node.
```
Comment (ppt-adsk) [inline test/lib/ufe/testGroupCmd.py:592]: "Create an xform"

File: test/lib/ufe/testGroupCmd.py [hunk 2]
```diff
@@ -581,6 +581,62 @@ def testGroupPreserveLoadRules(self):
...
+
+        # 1. Create a stage with a new layer.
+        # 2. Add a sub-layer.
+        psPathStr, psPath, psItem, layers = usdUtils.createLayeredStage(2)
+
+        topLayer = layers[1]
+        subLayer = layers[2]
+
+        # 3. Set the edit target to the new sub-layer.
+        # 4. Create on xform under the proxyShape node.
```
Comment (ppt-adsk) [inline test/lib/ufe/testGroupCmd.py:607]: "an xform" :)

File: test/lib/ufe/testGroupCmd.py [hunk 3]
```diff
@@ -581,6 +581,62 @@ def testGroupPreserveLoadRules(self):
...
+        # 6. Add a new prim under the xform, using the group command.
+        #    Note: we use an explicit create-group command because it is the only command
+        #          accessible from Python but the command as exposed by Maya requires a selection
+        #          so we create the command directly here to avoid that requirement.
+        aHier = ufe.Hierarchy.hierarchy(aItem)
+        if ufeUtils.ufeFeatureSetVersion() >= 3:
+            groupCmd = aHier.createGroupCmd(ufe.PathComponent("newGroup"))
+        else:
+            groupCmd = aHier.createGroupCmd(sn, ufe.PathComponent("newGroup"))
+        groupCmd.execute()
```
Comment (ppt-adsk) [inline test/lib/ufe/testGroupCmd.py:630]: Nit-pick: if you're going to call createGroupCmd() only to call execute() on the return command, may as well call createGroup() directly.

**Review by ppt-adsk — COMMENTED** [review]
Just a couple of nit-picks, otherwise looks good to me.  Thanks for doing this!

**Review by ppt-adsk — APPROVED** [review]

**Comment by pierrebai-adsk** [comment]
Note that the previous PF had passed and the only change afterward were typos in comments and executing the command immediately, which I tested locally. The PF system is currently having problems.

**Pull request merged**